define/types.go

```diff
@@ -173,11 +173,13 @@ func TempDirForURL(dir, prefix, url string) (name string, subdir string, err err
 	return "", "", errors.New("unreachable code reached")
 }
 
-func cloneToDirectory(url, dir string) ([]byte, string, error) {
+// parseGitBuildContext parses git build context to `repo`, `sub-dir`
+// `branch/commit`, accepts GitBuildContext in the format of
+// `repourl.git#branch/commit:subdir`.
```

Comment:
```suggestion
// `repourl.git[#[branch-or-commit]:subdir]`.
```
Optional change, to note that the subdir and branch-or-commit parts are optional.